app/models/ability.rb

```diff
@@ -1,9 +1,22 @@
-# frozen_string_literal: true
 class Ability
   include CanCan::Ability
 
   def initialize(user)
     user ||= User.new
+    if user.admin?
+      admin_abilities
+    else
+      user_abilities
+    end
+  end
+
```

Comment:
For this PR, I think only this should be enough:
```
def initialize(user)
  return unless user

  can :manage, User
end
```

But I do understand you are exploring the ways and keeping it for future references, so you can keep it if you want to for now. But in a future PR, you should use just as much is required for the current story.